### PR TITLE
Update README and versions for 1.43.0 / 24.08

### PR DESCRIPTION
**Pull request by pvijayakrish**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,114 +22,4 @@ limitations under the License.
 >
 > ##### LATEST RELEASE
```
Comment (mc-nv) [inline README.md:23]: Isn't this line redundant for "warning" message?

**Review by nv-kmcgill53 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (pvijayakrish) [reply README.md:23]: I was following the format done [before](https://github.com/triton-inference-server/model_analyzer/commit/7e2cf8d34a39b1aec55ed40751ab20893b3efcd8#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5). Please let me know if this needs to be updated.

Comment (mc-nv) [reply README.md:23]: @fpetrini15  for viz

File: README.md [hunk 2]
```diff
@@ -22,114 +22,4 @@ limitations under the License.
...
 >
-> You are currently on the `main` branch which tracks under-development progress towards the next release. <br>
-> The latest release of the Triton Model Analyzer is 1.42.0 and is available on branch
-> [r24.07](https://github.com/triton-inference-server/model_analyzer/tree/r24.07).
-
-Triton Model Analyzer is a CLI tool which can help you find a more optimal configuration, on a given piece of hardware, for single, multiple, ensemble, or BLS models running on a [Triton Inference Server](https://github.com/triton-inference-server/server/). Model Analyzer will also generate reports to help you better understand the trade-offs of the different configurations along with their compute and memory requirements.
```
[the remark this answers is not in the document]
Comment (mc-nv) [reply README.md:29]: It will be restored on release date

[the remark this answers is not in the document]
Comment (nv-braf) [reply README.md:29]: Thanks!

File: README.md (hunk 1 above)
Comment (fpetrini15) [reply README.md:23]: I also followed the format from [before](https://github.com/triton-inference-server/model_analyzer/pull/897/files). Unfortunately, it seems like we carry over issues. Hopefully, we can dedicate some time soon to streamlining this process.

**Pull request merged**